Kitodo/src/main/java/org/kitodo/production/interfaces/activemq/CreateNewProcessOrder.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.interfaces.activemq;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.jms.JMSException;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.kitodo.api.MdSec;
+import org.kitodo.api.Metadata;
+import org.kitodo.api.MetadataEntry;
+import org.kitodo.api.MetadataGroup;
+import org.kitodo.data.database.beans.ImportConfiguration;
+import org.kitodo.data.database.beans.Process;
+import org.kitodo.data.database.beans.Template;
+import org.kitodo.data.database.exceptions.DAOException;
+import org.kitodo.data.exceptions.DataException;
+import org.kitodo.exceptions.ProcessorException;
+import org.kitodo.production.dto.ProcessDTO;
+import org.kitodo.production.services.ServiceManager;
+import org.kitodo.production.services.data.ImportConfigurationService;
+
+/**
+ * Order to create a new process. This contains all the necessary data.
+ */
+public class CreateNewProcessOrder {
+
+    /* Catalog imports can be specified (none, one or more). An
+     * "importconfiguration" and a search "value" must be specified. The search
+     * is carried out in the default search field. If no hit is found, or more
+     * than one, the search aborts with an error message. In the case of
+     * multiple imports, a repeated import is carried out according to the
+     * procedure specified in the rule set. */
+    private static final String FIELD_IMPORT = "import";
+    private static final String FIELD_IMPORT_CONFIG = "importconfiguration";
+    private static final String FIELD_IMPORT_VALUE = "value";
+
+    /* Additionally metadata can be passed. Passing multiple metadata or passing
+     * grouped metadata is also possible. */
+    private static final String FIELD_METADATA = "metadata";
+
+    /* A parent process can optionally be specified. The process ID or the
+     * process title can be specified. (If the value is all digits, it is
+     * considered the process ID, else it is considered the process title.) The
+     * process must be found in the client’s processes. If no parent process is
+     * specified, but a metadata entry with a use="higherLevelIdentifier" is
+     * included in the data from the catalog, the parent process is searched for
```

Comment:
Is it really enough to provide a metadata import with a `higherLevelIdentifier`? In case a import configuration is provided, the import is explicitely called with an import level of 1 (IMPORT_WITHOUT_ANY_HIERARCHY):

```java

 List<TempProcess> processHierarchy = importService.importProcessHierarchy(
                order.getImports().get(which).getValue(), order.getImports().get(which).getKey(),
                order.getProjectId(), order.getTemplateId(), IMPORT_WITHOUT_ANY_HIERARCHY,
                rulesetManagement.getFunctionalKeys(FunctionalMetadata.HIGHERLEVEL_IDENTIFIER));
```

This leads to the code for loading the parent to actually never be called.
https://github.com/kitodo/kitodo-production/blob/master/Kitodo/src/main/java/org/kitodo/production/services/data/ImportService.java#L552-L557

So we are not creating any parents as described in the spec, but we are also not searching for any parent by the `recordIdentifier`.